File: cs50-rust/ps0-04-einstein/src/main.py
def main():
    print()
    print("enter mass in kg (rounded to nearest kg): ")
    mass = int(input())
    print()
    print(f"energy = {energy(mass)} joules");


def energy(mass: int) -> int:
    SPEED_OF_LIGHT_MS = 299792458
    return mass * SPEED_OF_LIGHT_MS


if __name__ == "__main__":
    main()

File: cs50-rust/ps0-04-einstein/src/main.rs
use std::io::stdin;

fn main() {
    println!();
    println!("enter mass in kg (rounded to nearest kg): ");
    let mass = get_mass();
    println!();
    println!("energy = {} kilojoules", energy(mass));
}

fn get_mass() -> u64 {
    let mut user_input = String::new();
    stdin()
        .read_line(&mut user_input)
        .expect("Error reading line");
    user_input
        .trim()
        .parse::<u64>()
        .expect("Error, whole number not entered!")
}

fn energy(mass: u64) -> u64 {
    const SPEED_OF_LIGHT_MS: u64 = 299_792_458;
    (mass * SPEED_OF_LIGHT_MS) / 1000
}
